### PR TITLE
Zwave: add support for the Meter Tbl Monitor Command Class

### PR DESCRIPTION
**Pull request by jongj**
Add support for the METER_TBL_CURRENT_DATA_REPORT message.
In this message the current meter value is reported.
The command class also supports historical data, this is not
supported (yet).

Signed-off-by: Jorg de Jong <jorg@dejong.info> (github: jongj)

### CODE REVIEW

File: addons/binding/org.openhab.binding.zwave/src/main/java/org/openhab/binding/zwave/internal/converter/ZWaveMeterTblMonitorConverter.java [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * ZWaveMeterTblMonitorConverter class. Converter for communication with the {@link ZWaveMeterTblMonitorCommandClass}.
+ * Implements polling of the sensor status and receiving of sensor events.
+ *
+ * @author Jorg de Jong
+ */
+public class ZWaveMeterTblMonitorConverter extends ZWaveCommandClassConverter {
```
Comment (cdjackson) [inline addons/binding/org.openhab.binding.zwave/src/main/java/org/openhab/binding/zwave/internal/converter/ZWaveMeterTblMonitorConverter.java]: Please add the XStream annotation

Comment (cdjackson) [inline addons/binding/org.openhab.binding.zwave/src/main/java/org/openhab/binding/zwave/internal/converter/ZWaveMeterTblMonitorConverter.java]: Sorry - I thought this was the command class - not the converter... Disregard this comment...

File: addons/binding/org.openhab.binding.zwave/src/main/java/org/openhab/binding/zwave/internal/converter/ZWaveMeterTblMonitorConverter.java [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+     */
+    @Override
+    public State handleEvent(ZWaveThingChannel channel, ZWaveCommandClassValueEvent event) {
+        String meterScale = channel.getArguments().get("type");
+        ZWaveMeterTblMonitorValueEvent meterEvent = (ZWaveMeterTblMonitorValueEvent) event;
+        // logger.debug("Meter converter: scale {} <> {}", meterScale, meterEvent.getMeterScale());
+
+        // Don't trigger event if this item is bound to another sensor type
+        if (meterScale != null && MeterTblMonitorScale.getMeterScale(meterScale) != meterEvent.getMeterScale()) {
+            logger.debug("Not the right scale {} <> {}", meterScale, meterEvent.getMeterScale());
```
Comment (cdjackson) [inline addons/binding/org.openhab.binding.zwave/src/main/java/org/openhab/binding/zwave/internal/converter/ZWaveMeterTblMonitorConverter.java]: I would suggest removing this - I guess you copied it from the meter converter, and that was just there for some initial debugging and will be removed soon too.

**Comment by cdjackson** [comment]
Thanks. Looks good... Can I ask you to add a few test cases for the converter and command class?

**Comment by jongj** [comment]
Thanks, I started with a unit test:
```
    @Test
    public void Gas_Cubic_Meters() {

        byte[] packetData = { 0x01, 0x26, 0x00, 0x04, 0x00, 0x07, 0x20, 0x56, 0x01, 0x60, 0x0D, 0x01, 0x01, 0x3D, 0x0D,
                0x00, 0x01, 0x03, 0x00, 0x00, 0x00, 0x00, 0x01, 0x01, 0x01, 0x1B, 0x19, 0x60, 0x00, 0x00, 0x02, 0x6C,
                0x68, 0x00, 0x00, 0x00, 0x00, (byte) 0xA6, (byte) 0xEC, (byte) 0xDD };

        List<ZWaveEvent> events = processMessage(packetData);

        assertEquals(events.size(), 1);

        ZWaveMeterTblMonitorValueEvent event = (ZWaveMeterTblMonitorValueEvent) events.get(0);

        assertEquals(event.getCommandClass(), CommandClass.METER_TBL_MONITOR);
        assertEquals(event.getEndpoint(), 1);
        assertEquals(event.getMeterScale(), ZWaveMeterTblMonitorCommandClass.MeterTblMonitorScale.G_Cubic_Meters);
        assertEquals(event.getMeterType(), ZWaveMeterTblMonitorCommandClass.MeterTblMonitorType.GAS);
        assertEquals(event.getValue(), new BigDecimal("0.620"));
    }
```
But got stuck on the combination of  CRCEncap with MultiInstance, can you help with that?

**Comment by cdjackson** [comment]
I’d recommend trying to produce a test that doesn’t use the encapsulation classes if possible - thereby just testing this class and not the other classes. If you really must use multi instance, then there’s a separate package for that -:
package org.openhab.binding.zwave.test.internal.protocol.commandclass.multiinstance;

This extends a class that takes care of handling the multi instance stuff, but I’ve not done anything for the CRC encap yet…

Really though, I’d recommend to only test a single command class at a time so that we know where any errors are when doing regression.

Thanks.

**Comment by jongj** [comment]
I added two unit tests, and removed the debug comment. 
The build is failing on a !max binding test, same as the regular master build from last night.

**Comment by cdjackson** [comment]
This all looks good - I'll merge once the current PR is sorted and the clock tests pass again...

**Pull request merged**